File: R/create_data_list.R
#' Creates a 2-levels list of urls, pointing to open source data
#' 
#' @param source_file yaml file containing data urls 
#' @return list (level 1 = concepts, level 2 = year).
#'
#' @examples
#'  create_data_list("sources.yml")
#'  
create_data_list <- function(source_file){
  catalogue <- yaml::read_yaml(source_file)
  return(catalogue)
}

File: correction/src/create_data_list.py
import yaml

def create_data_list(source_file):
    """
    Reads a YAML file and returns the contents as a dictionary.

    Args:
        source_file (str): The path to the YAML file.

    Returns:
        dict: The contents of the YAML file.
    """
    with open(source_file, 'r') as file:
        catalogue = yaml.safe_load(file)
    return catalogue
